File: server/router/campus/campus_announcement.go
package campus

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type CampusAnnouncementRouter struct{}

func (r *CampusAnnouncementRouter) InitCampusAnnouncementRouter(Router *gin.RouterGroup) {
	announcementRouter := Router.Group("campusAnnouncement").Use(middleware.OperationRecord())
	announcementRouterWithoutRecord := Router.Group("campusAnnouncement")
	{
		announcementRouter.POST("createCampusAnnouncement", campusAnnouncementApi.CreateCampusAnnouncement)
		announcementRouter.PUT("updateCampusAnnouncement", campusAnnouncementApi.UpdateCampusAnnouncement)
		announcementRouter.DELETE("deleteCampusAnnouncement", campusAnnouncementApi.DeleteCampusAnnouncement)
		announcementRouter.POST("updateCampusAnnouncementStatus", campusAnnouncementApi.UpdateCampusAnnouncementStatus)
	}
	{
		announcementRouterWithoutRecord.GET("getCampusAnnouncementList", campusAnnouncementApi.GetCampusAnnouncementList)
		announcementRouterWithoutRecord.GET("findCampusAnnouncement", campusAnnouncementApi.FindCampusAnnouncement)
	}
}

File: web/src/api/campusAnnouncement.js
import service from '@/utils/request'

export const getCampusAnnouncementList = (params) => {
  return service({
    url: '/campusAnnouncement/getCampusAnnouncementList',
    method: 'get',
    params
  })
}

export const findCampusAnnouncement = (params) => {
  return service({
    url: '/campusAnnouncement/findCampusAnnouncement',
    method: 'get',
    params
  })
}

export const createCampusAnnouncement = (data) => {
  return service({
    url: '/campusAnnouncement/createCampusAnnouncement',
    method: 'post',
    data
  })
}

export const updateCampusAnnouncement = (data) => {
  return service({
    url: '/campusAnnouncement/updateCampusAnnouncement',
    method: 'put',
    data
  })
}

export const deleteCampusAnnouncement = (data) => {
  return service({
    url: '/campusAnnouncement/deleteCampusAnnouncement',
    method: 'delete',
    data
  })
}

export const updateCampusAnnouncementStatus = (data) => {
  return service({
    url: '/campusAnnouncement/updateCampusAnnouncementStatus',
    method: 'post',
    data
  })
}

